src/common/telemetry/src/metric.rs

```diff
@@ -25,3 +33,233 @@ pub fn dump_metrics() -> Result<String, String> {
         .map_err(|_| "Encode metrics failed".to_string())?;
     String::from_utf8(buffer).map_err(|e| e.to_string())
 }
+
+/// `MetricFilter` used in `report_metric_task`.
+/// for metric user don't want collect, return a `false`, else return a `true`
+pub type MetricFilter = Box<dyn Fn(&MetricFamily) -> bool>;
+
+pub fn report_metric_task(config: RemoteWriteOptions) {
+    if !config.enable {
+        return;
+    }
+    if config.write_interval.as_secs() == 0 {
+        panic!("Expected Remote write write_interval greater than zero")
+    }
+    let _ = std::thread::spawn(move || loop {
+        std::thread::sleep(config.write_interval);
+        match report_metric(&config.endpoint, None) {
+            Ok(resp) => {
+                if !resp.status().is_success() {
+                    tracing::error!("report metric in remte write error, msg: {:#?}", resp);
+                }
+            }
+            Err(e) => tracing::error!("report metric in remte write failed, error {:#?}", e),
+        };
+    });
+}
+
+/// Export the collected metrics, encode metrics into [RemoteWrite format](https://prometheus.io/docs/concepts/remote_write_spec/),
+/// and send metrics to Prometheus remote-write compatible receiver (e.g. `greptimedb`) specified by `url`.
+/// User could use `MetricFilter` to filter metric they don't want collect
+/// This function write in sync way, **don't use in async function**
+pub fn report_metric(url: &str, filter: Option<&MetricFilter>) -> Result<Response, String> {
```

Comment:
cc @shuiyisong 

We plan to provide this `MetricFilter` as a `Plugin`, allowing others to opt-in/opt-out some unnecessary or sensitive metrics, PTAL

Reply:
We can build a config option for metric filter, load it into plugins upon startup, and get filters from plugins later on(like here).

The problem is that `common_telemetry::init_global_logging` get invoked in `main()`, where building plugins happens during initialization of applications(Frontend, Datanode, etc.). We might need to figure out a way to pass plugins into this function call.

Reply:
It seems standalone lacks of `Plugin` mechanism, if we move `MetricFilter` as a `Plugin`, it's difficult to align behavior of all components